subprojects/model-core/src/main/java/org/gradle/api/internal/provider/sources/process/DefaultExecOutput.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.provider.sources.process;
+
+import org.gradle.api.Transformer;
+import org.gradle.api.internal.provider.sources.process.ProcessOutputValueSource.ExecOutputData;
+import org.gradle.api.provider.Provider;
+import org.gradle.process.ExecOutput;
+import org.gradle.process.ExecResult;
+
+import java.io.Serializable;
+import java.nio.charset.Charset;
+
+public class DefaultExecOutput implements ExecOutput {
+    private final Provider<ExecOutputData> dataProvider;
+
+    public DefaultExecOutput(Provider<ExecOutputData> dataProvider) {
+        this.dataProvider = dataProvider;
+    }
+
+    @Override
+    public Provider<ExecResult> getResult() {
+        return dataProvider.map((Transformer<ExecResult, ExecOutputData> & Serializable) ExecOutputData::getResult);
```

Comment:
💄 could use `SerializableLambdas.transformer` here, e.g., `dataProvider.map(transformer(ExecOutputData::getResult))`

Reply:
I was a bit reluctant to move `SerializableLambdas` out of `core` (which isn't accessible there) but I think having it in `base-services` alongside functional interfaces that it decorates should be fine in the end.